packages/renderer/src/lib/compose/ComposeDetailsLogs.svelte

```diff
@@ -0,0 +1,165 @@
+<script lang="ts">
+import { onDestroy, onMount } from 'svelte';
+import { Terminal } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit';
+import 'xterm/css/xterm.css';
+import { TerminalSettings } from '../../../../main/src/plugin/terminal-settings';
+import { getPanelDetailColor } from '../color/color';
+import { isMultiplexedLog } from '../stream/stream-utils';
+import EmptyScreen from '../ui/EmptyScreen.svelte';
+import NoLogIcon from '../ui/NoLogIcon.svelte';
+import type { ComposeInfoUI } from './ComposeInfoUI';
+import { ansi256Colours, colourizedANSIContainerName } from '../editor/editor-utils';
+
+export let compose: ComposeInfoUI;
+
+// Logging element
+let logsXtermDiv: HTMLDivElement;
+let refCompose;
+
+// Log initialization
+let noLogs = true;
+let logsTerminal: Terminal;
+
+$: {
+  if (refCompose && refCompose.status !== compose.status) {
+    logsTerminal?.clear();
+    fetchComposeLogs();
+  }
+  refCompose = compose;
+}
+
+// Terminal resize
+let resizeObserver: ResizeObserver;
+let termFit: FitAddon;
+let currentRouterPath: string;
+
+// Router path for the logging
+let logsRouterPath = `/compose/${encodeURI(compose.name)}/${encodeURI(compose.engineId)}/logs`;
+
+// Create a map that will store the ANSI 256 colour for each container name
+// if we run out of colours, we'll start from the beginning.
+const colourizedContainerName = new Map<string, string>();
+
+// Callback for logs which will output the logs to the terminal
+function callback(name: string, data: string) {
```

Comment:
Maybe rename to have a more meaningful name lilke `onLogMessage` or `onLogData` ?

Reply:
copied-and-pasted from the podman logs section, I'd rather keep the same name.